### PR TITLE
New Explore DMR Codelab

### PR DESCRIPTION
**Pull request by Chaymee**
In this new codelab we explore how to create a DMR links between Solace Brokers using the WebUI to create a basic Event Mesh.  Then we explore two simple methods to publishing events into the Event mesh and subscribing to events on remote brokers to "watch" events flow across DMR links in the Event mesh.  

This is the first codelab that I have contributed and I do not believe it is in its final form yet.  I would love feedback on how to improve the lab. Specifically I am worried about length, number and format of screenshots, use of notes and links to other material.  Finaly how to asses the Duration of each section.

-Jamieson

### CODE REVIEW

File: markdown/explore-dmr/explore-dmr.md [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+# DMR (Event Mesh) Exploration Procedures
+
+## Objectives
+Duration: 0:05:00
+
+### Prerequisites
+1. 2+ Solace PubSub+ Brokers deployed in environments where they are visable to eachother over a network. 
+2. Access to the Solace Broker WebUI for all Brokers via administrator credentials. 
+
+<br>**⚠️ A note on security: The brokers must be secured with a TLS certificate. If you are using a Solace PubSub+ Docker image please check out [my blog on quickly securing the broker with a self-signed TLS certificate.](SomeblogIhaveNotWrittenYet)**
```
Comment (TamimiGitHub) [inline markdown/explore-dmr/explore-dmr.md:20]: Once blog is up we will update codelab here

File: markdown/explore-dmr/explore-dmr.md [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+
+If you do not meet both of these requirements please use the following resources to setup and configure Solace PubSub+ Brokers.
+* [Solace Broker CodeLab](https://codelabs.solace.dev/codelabs/get-started-basics/index.html)
+* [Solace Cloud Trial](https://console.solace.cloud/login/new-account)
+* [Solace PubSub+ Standard Docker](https://hub.docker.com/r/solace/solace-pubsub-standard)
```
Comment (TamimiGitHub) [inline markdown/explore-dmr/explore-dmr.md:25]: I would recommend pointing to this url instead 
https://solace.com/products/event-broker/software/getting-started/

File: markdown/explore-dmr/explore-dmr.md [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+<br>
+
+In this post will lay the groundwork for a Solace Event Mesh by setting up a Dynamic Message Routing (DMR) link between two stand alone brokers.  These brokers can be running in Docker or on Solace Cloud, in any cloud environment.  Then we will explore how events are propogatted across a DMR link.  Setup publishers and Subscribers to each of the brokers in our Event Mesh to see events publish to one broker be dynamically routed to the other brokers in our Event Mesh where there are active subscribers.
```
Comment (TamimiGitHub) [inline markdown/explore-dmr/explore-dmr.md:28]: Typo? 
> This codelab will lay the ....

File: markdown/explore-dmr/explore-dmr.md [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+
+12. Successful DMR Link:
+    After communication tests we will have a successful DMR Link.  
+    <br>⚠️ If you encounter an error please review a DMR troubleshooting steps to determine how to correct the error and build a successful DMR Link between brokers
+    ![Successful DMR Link](img/successfulLink.png)
+
+**❗️❗️♻️ At this point if you would like to include more than 2 brokers in your Event Mesh repeat steps 6 to 12 for each additional broker you wish to include in the mesh.  <br>⚠️Note that events will not traverse N brokers to reach a desired broker so you have to form a direct DMR Link between each and every broker for form a full Event Mesh**
+
+### Conclusion
+In the prevous steps we have provisioned a cluster on each broker to prepare them to participate in an Event Mesh.  Then we created a DMR Link between the brokers to form the Event Mesh.  Events will no be intelligently and Dynamically Routed from the Solace PubSub+ brokers where they are produced to the broker where there are interested subscribers. Continue to learn how we can explore the behavior of events over a DMR Link.
```
Comment (TamimiGitHub) [inline markdown/explore-dmr/explore-dmr.md:116]: Typo? events will not or they will? 
> Events will **not** be intelligently and.....

File: markdown/explore-dmr/explore-dmr.md [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+* Publish directly to queues
+* Subscribe to Solace topics
+* Subscribe to Solace queues
+* Subscribe to Solace topics using Solace wild cards
+* Experience Solace Broker translate different message protocols (requires the use of Try Me and an additional publisher or subscriber).
+
+<br>
+
+**In the next section we will explore how using the Try Me tool to Publish and Subscribe only to a single individual Solace Broker will allow us to obtain events from any accessible topic published to any of the Solace Brokers connected to our Solace DMR Event Mesh.** 
+<br>🆘!!!Maybe here I should include some topic architecture for 2 brokers that will allow you to see the DMR Event Mesh in action?
```
Comment (TamimiGitHub) [inline markdown/explore-dmr/explore-dmr.md:137]: sos! 🆘

File: markdown/explore-dmr/explore-dmr.md [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+
+## Setting up and running "Try Me"
+Duration 00:10:00
```
Comment (TamimiGitHub) [inline markdown/explore-dmr/explore-dmr.md:140]: I believe this should be `Duration 0:10:00` you have an extra zero at the beginning which broke errythang

File: markdown/explore-dmr/explore-dmr.md [hunk 7]
```diff
@@ -0,0 +1,295 @@
...
+6. Select the “Connect” button on the Subscriber side of “Try Me” as well.  The connection details will carry over from the Publisher side unless you expand the drop down menu, uncheck the “Same as Publisher” option and modify them.
+    
+    ![Untitled](img/Untitled_5.png)
+    
+7.  Now we can customize our Publish topic, Delivery Mode and Message Content and Publish messages directly to our local (on-premise) broker. 
+    
+    ![Untitled](img/Untitled_6.png)
+    
+8. We can also subscribe to any topics that we would like on the right side by changing the subscribe field and pressing “Subscribe”. The topic can be one that is published to our local broker or published to any other broker in the DMR(Event Mesh) that your user has access to.
+Do not forget about our Solace wild cards here
```
Comment (TamimiGitHub) [inline markdown/explore-dmr/explore-dmr.md:173]: I recommend linking to https://docs.solace.com/Messaging/Wildcard-Charaters-Topic-Subs.htm here

File: markdown/explore-dmr/explore-dmr.md (hunk 1 above)
Comment (Chaymee) [reply markdown/explore-dmr/explore-dmr.md:20]: I think I will pull this out as a separate PR. I have not had a chance to write up self signed cert creation for the broker.

**Pull request closed without merging**